src/kiwixapp.cpp

```diff
@@ -92,6 +92,15 @@ void KiwixApp::init()
             this->openZimFile(message);
         }
     });
+    connect(&m_watcher, &QFileSystemWatcher::directoryChanged, this, [=](QString monitorDir) {
+        m_library.asyncLoadMonitorDir(monitorDir);
+    });
+    QString monitorDir = m_settingsManager.getMonitorDir();
+    if (monitorDir != "") {
+        m_library.setMonitorDirZims(m_library.getLibraryZimsFromDir(monitorDir));
+        m_library.asyncLoadMonitorDir(monitorDir);
+        m_watcher.addPath(monitorDir);
```

Comment:
With the current order of these two lines there is a small chance that a file appears or is deleted in the monitor directory after its contents has been read but before it started being watched.